File: app/src/main/java/com/serwylo/lexica/activities/score/ScoreTabViewHolder.java
package com.serwylo.lexica.activities.score;

import android.widget.FrameLayout;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.serwylo.lexica.game.Game;

class ScoreTabViewHolder extends RecyclerView.ViewHolder {

    private final ScoreActivity activity;
    private final FrameLayout parent;

    ScoreTabViewHolder(@NonNull ScoreActivity activity, @NonNull FrameLayout parent) {
        super(parent);
        this.activity = activity;
        this.parent = parent;
    }

    void bindFoundWords(@NonNull Game game) {
        new FoundWordsViewBinder(activity, parent, game);
    }

    void bindMissedWords(@NonNull Game game) {
        new MissedWordsViewBinder(activity, parent, game);
    }
}

File: app/src/main/java/com/serwylo/lexica/activities/score/ScoreTabViewHolder.kt
package com.serwylo.lexica.activities.score

import android.widget.FrameLayout
import androidx.recyclerview.widget.RecyclerView
import com.serwylo.lexica.game.Game

internal class ScoreTabViewHolder(private val activity: ScoreActivity, private val parent: FrameLayout) : RecyclerView.ViewHolder(parent) {
    fun bindFoundWords(game: Game, sorter: Sorter) {
        FoundWordsViewBinder(activity, parent, game, sorter)
    }

    fun bindMissedWords(game: Game, sorter: Sorter) {
        MissedWordsViewBinder(activity, parent, game, sorter)
    }

    fun bindNextRound(game: Game) {
        NextRoundViewBinder(activity, parent, game)
    }
}
